rdk/rdk.py

```diff
@@ -1337,6 +1552,340 @@ def deploy(self):
+                source_periodic = rule_params['SourcePeriodic']
+
+            combined_input_parameters = {}
+            if 'InputParameters' in rule_params:
+                combined_input_parameters.update(json.loads(rule_params['InputParameters']))
+
+            if 'OptionalParameters' in rule_params:
+                #Remove empty parameters
+                keys_to_delete = []
+                optional_parameters_json = json.loads(rule_params['OptionalParameters'])
+                for key, value in optional_parameters_json.items():
+                    if not value:
+                        keys_to_delete.append(key)
+                for key in keys_to_delete:
+                    del optional_parameters_json[key]
+                combined_input_parameters.update(optional_parameters_json)
+
+            if 'SourceIdentifier' in rule_params:
+                print("Found Managed Rule.")
+                #create CFN Parameters for Managed Rules
+
+                try:
+                    rule_description = rule_params["Description"]
+                except KeyError:
+                    rule_description = rule_name
+                my_params = [
+                    {
+                        'ParameterKey': 'RuleName',
+                        'ParameterValue': rule_name,
+                    },
+                    {
+                        'ParameterKey': 'Description',
+                        'ParameterValue': rule_description,
+                    },
+                    {
+                        'ParameterKey': 'SourceEvents',
+                        'ParameterValue': source_events,
+                    },
+                    {
+                        'ParameterKey': 'SourcePeriodic',
+                        'ParameterValue': source_periodic,
+                    },
+                    {
+                        'ParameterKey': 'SourceInputParameters',
+                        'ParameterValue': json.dumps(combined_input_parameters),
+                    },
+                    {
+                        'ParameterKey': 'SourceIdentifier',
+                        'ParameterValue': rule_params['SourceIdentifier']
+                    }]
+                my_cfn = my_session.client('cloudformation')
+
+                #deploy config rule
+                cfn_body = os.path.join(path.dirname(__file__), 'template',  "configManagedRuleOrganization.json")
+
+                try:
+                    my_stack_name = self.__get_stack_name_from_rule_name(rule_name)
+                    my_stack = my_cfn.describe_stacks(StackName=my_stack_name)
+                    #If we've gotten here, stack exists and we should update it.
+                    print ("Updating CloudFormation Stack for " + rule_name)
+                    try:
+                        cfn_args = {
+                            'StackName': my_stack_name,
+                            'TemplateBody': open(cfn_body, "r").read(),
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *open*. The resource is allocated by call *builtins.open*. Currently, there are execution paths that do not contain closure statements, for example, when *builtins.len()* throws an exception. To prevent this resource leak, close the object returned by *open()* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).